### PR TITLE
Fix #46 and make backward-compatible with old style computed table

### LINKED ISSUE #46: Erroneous cascading delete on non-congruent non-pk foreign refs
Follow set of tables lead to incorrect cascading delete:

Image
```matlab
%{
test.Image (manual) # collection of all image data
image_id          :int          #unique image id
-----
image_data=null   :longblob     #image data
%}
classdef Image < dj.Relvar
end
```

MonkeyImage
```matlab
%{
test.MonkeyImage (manual) # collection of monkey images
subj_id         :int      #unique id of the monkey
-----
-> test.Image
%}
classdef MonkeyImage < dj.Relvar
end
```

ModifiedMonkeyImage
```matlab
%{
test.ModifiedMonkeyImage (manual) # collection of modified monkey images
-> test.MonkeyImage
modification_type    :varchar(255)      #type of modification
-----
-> test.Image
%}
classdef ModifiedMonkeyImage < dj.Relvar
end
```

In the above three tables, `Image` is where all the images are actually stored as `longblob`, with each image given a unique `image_id`. `MonkeyImage` is a convenient collection of all images for monkeys, with unique entry for each monkey identified by `subj_id`. Finally, `ModifiedMonkeyImage` is a collection of modified images for the monkey, uniquely identified by the `subj_id` of the monkey as the `modification_type` descriptor. Note that although `ModifiedMonkeyImage` is a child of `MonkeyImage` and that both have non-pk foreign ref back to `Image`, they will point to *different image entries*!

With the three tables defined as above, the follow code illustrates the issue:

```matlab
% insert two images, one for monkey unmodified, and one for cropped monkey image
insert(test.Image, struct('image_id', 1);
insert(test.Image, struct('image_id', 2);

% insert the image under MonkeyImage
insert(test.MonkeyImage, struct('subj_id', 100, 'image_id', 1);

% insert the modified image for "subj_id = 100", where "crop" has been applied
% note that here the image_id=2, not 1!
insert(test.ModifiedMonkeyImage, struct('subj_id', 100, 'modification_type', 'crop', 'image_id', 2);

% now try to delete "image_id=2" from "Image", this should cause cascading delete in 
% test.ModifiedMonkeyImage

del(test.Image & 'image_id =  2')

% OUTPUT BELOW
ABOUT TO DELETE:
       1 tuples from `edgar_sandbox`.`image` (manual)

Proceed to delete? (yes/no) > yes
Deleting from test.Image

 ** delete rolled back due to to error
Error using mym
Cannot delete or update a parent row: a foreign key constraint fails (`edgar_sandbox`.`modified_monkey_image`, CONSTRAINT
`modified_monkey_image_ibfk_2` FOREIGN KEY (`image_id`) REFERENCES `image` (`image_id`) ON UPDATE CASCADE)
```

This issue has been mentioned on datajoint/datajoint-python#15 and I am not sure if it has been appropriately dealt with in datajoint-python either. This is caused by an inappropriate cascading when there exists multiple paths from a parent table (e.g. `Image`) to the target (`ModifiedMonkeyImage`, here paths are `Image -> ModifiedMonkeyImage` and `Image -> MonkeyImage -> ModifiedMonkeyImage`).

### PR DESCRIPTION
**Pull request by eywalker**
* Fix #46 - cascading delete with non-congruent FK paths are handled correctly
* fix backward-compatibility with old computed table declaration

### CODE REVIEW

File: +dj/Relvar.m [hunk 1]
```diff
@@ -64,22 +64,16 @@ function cleanup(self)
                 rels = [rels{:}];
                 rels(1) = rels(1) & self.restrictions;
                 
-                % apply proper restrictions
```
Comment (dimitri-yatsenko) [inline +dj/Relvar.m]: What happened to these restrictions?

File: +dj/+internal/Table.m [hunk 2]
```diff
@@ -632,7 +632,7 @@ function create(self)
             % parse table schema, name, type, and comment
             switch true
                     
-                case {isa(self, 'dj.internal.UserRelation'), isa(self, 'dj.Part'), isa(self, 'dj.Jobs')}
+                case {isa(self, 'dj.internal.UserRelation') && (~isa(self, 'dj.internal.AutoPopulate') || isa(self, 'dj.Computed')), isa(self, 'dj.Part'), isa(self, 'dj.Jobs')}
```
Comment (dimitri-yatsenko) [inline +dj/+internal/Table.m]: what about `dj.Imported`

File: +dj/Relvar.m (hunk 1 above)
Comment (eywalker) [reply +dj/Relvar.m]: this is absorbed by the restriction with `pro(rel(i))`

File: +dj/+internal/Table.m (hunk 2 above)
Comment (eywalker) [reply +dj/+internal/Table.m]: yep, forgot about it. Will add.

File: +dj/Relvar.m (hunk 1 above)
Comment (dimitri-yatsenko) [reply +dj/Relvar.m]: This is inefficient and for now I must turned down this solution.

Comment (eywalker) [reply +dj/Relvar.m]: Agreed - I would rather implement much more full fledged solution. Will go ahead and close this PR for now.

**Pull request closed without merging**